### DISCUSSION TITLE
Chat Disappear After Close the Browser

### DISCUSSION THREAD

**Discussion by senja24** [lobehub/lobehub#1463] [General Question]
I encountered an issue where the chat history and all settings disappear when I close the browser. Unfortunately, all previous chat history is missing as well. Has anyone else experienced this problem or found a solution for it? Its included api key

**Comment by arvinxx** [comment]
@senja24 if you are using your browser in anonymous mode?

**Reply by senja24** [reply]
No, just regular mode in chrome. I'm facing this on chrome windows, linux and on my android phone

**Reply by arvinxx** [reply]
check whether your just miss the topics? https://chat-docs.lobehub.com/en/usage/agents/concepts#topics-and-assistants

**Reply by senja24** [reply]
I'm understand about topics and agents, but i mean sometimes it's just looks like deleting localdatabase of lobe chat. I'm not deploying my own, but still use chat-preview lobehub.

When facing this issue, its just like i starting new one in browser. So i need to put my access key again and installing the plugin also. Because all of those is disappeared.

**Reply by arvinxx** [reply]
> When facing this issue, its just like i starting new one in browser

Yes because different browser don't share each local databsae

**Reply by senja24** [reply]
Yeps, i understand this. But this issue is in one browser. I have already test it and got the issue.

Im trying on chrome, its working fine. But sometimes when i close tab and close chrome app, my configuration is gone, also all chat history.

**Reply by CXwudi** [reply]
Or maybe did you have any security software that periodically delete all cookies when the browser is closed. For example, CCleaner has the option to clear all cookies upon closing the browser and you probably don't want that

**Reply by senja24** [reply]
I never installed something like that. Other project which is using localstorage is working fine. I'm facing this issue when testing this project.

**Reply by arvinxx** [reply]
@senja24 is your disk going to be full?

**Reply by senja24** [reply]
On my phone device is have around 3gb free space, I'm using android with chrome as browser. But for my laptop i think got enough space, more than 40gb and using chrome also. And both of them got same issue. Isn't it enough?

I try using another app like nextchat, its working fine. My history chat not missing after close the browser and tab.

**Reply by arvinxx** [reply]
we are using indexedDB in the browser. Here is a strategy you can refer to: https://developer.mozilla.org/en-US/docs/Web/API/Storage_API/Storage_quotas_and_eviction_criteria#storage_pressure_eviction

IndexedDB don't depend on the absolute enough space, but the relative percent. 

please run this code in your inspector to check whether your browser is persisting the indexedDB

```js
if (navigator.storage && navigator.storage.persist) {
  navigator.storage.persisted().then((persistent) => {
    if (persistent) {
      console.log("Storage will not be cleared except by explicit user action");
    } else {
      console.log("Storage may be cleared by the UA under storage pressure.");
    }
  });
}
```


If the result is not persistent, you can run this code to force persist the indexedDB database as a temporary fixture, and I will fix it soon

```js
if (navigator.storage && navigator.storage.persist) {
  navigator.storage.persist().then((persistent) => {
    if (persistent) {
      console.log("Storage will not be cleared except by explicit user action");
    } else {
      console.log("Storage may be cleared by the UA under storage pressure.");
    }
  });
}
```

**Reply by senja24** [reply]
yes, i got "Storage may be cleared by the UA under storage pressure." its that mean that the problem is on the device?

**Reply by arvinxx** [reply]
yes. Do you still meet the same problem when using the second code?

**Reply by senja24** [reply]
Yes, same problem

**Reply by senja24** [reply]
![image](https://github.com/lobehub/lobe-chat/assets/33771197/42249bbe-3ef8-4b91-b93d-4815f028c5ad)
This the result. Same

**Reply by arvinxx** [reply]
Have you any config with privacy to stop cookie or site data storage?

**Reply by senja24** [reply]
I don't think so. It's also happened on Brave browser (android)

**Reply by arvinxx** [reply]
Maybe I need to dive into more to figure out reason.
